test/SCBridgeWallet.test.ts

```diff
@@ -75,6 +78,100 @@ describe("SCBridgeWallet", function () {
     };
   }
 
+  it("should support executing a simple L1 transfer through the entrypoint using a counterfactual deployment", async function () {
+    const owner = ethers.Wallet.createRandom();
+    const intermediary = ethers.Wallet.createRandom();
+
+    const entryPointDeployer = await ethers.getContractFactory("EntryPoint");
+    const entrypoint = await entryPointDeployer.deploy();
+    const entrypointAddress = await entrypoint.getAddress();
+
+    const salt = ethers.encodeBytes32String("Super secret salt");
+
+    const factoryDeployer = await hre.ethers.getContractFactory(
+      "SCBridgeAccountFactory",
+    );
+    const scwFactory =
+      (await factoryDeployer.deploy()) as unknown as SCBridgeAccountFactory;
+
+    // Call the factory function to compute the address of the SCW
+    // This can be done locally but it's easier to call into a view function
+    const precomputedSCWAddress = await scwFactory.computeAddress(
+      owner.address,
+      intermediary.address,
+      entrypointAddress,
+      salt,
+    );
+
+    await fundAddresses(
+      [owner.address, intermediary.address, precomputedSCWAddress],
+      ethers.parseEther("1.0"),
+    );
+
+    // The entrypoint contract requires a deposit from the submitter if not using a paymaster
+    await entrypoint.depositTo(precomputedSCWAddress, {
+      value: ethers.parseEther("1.0"),
+    });
+    const n = await ethers.provider.getNetwork();
+
+    // Generate a random payee address that we can use for the transfer.
+    const payee = ethers.Wallet.createRandom();
+
+    // Encode calldata that calls the execute function to perform a simple transfer of ether to the payee.
+    const callData =
+      SCBridgeWallet__factory.createInterface().encodeFunctionData("execute", [
+        payee.address,
+        ethers.parseEther("0.5"),
+        "0x",
+      ]);
+
+    const initCode = ethers.concat([
+      await scwFactory.getAddress(),
+      scwFactory.interface.encodeFunctionData("createAccount", [
+        owner.address,
+        intermediary.address,
+        entrypointAddress,
+        salt,
+      ]),
+    ]);
+
+    const userOp: UserOperationStruct = {
+      sender: precomputedSCWAddress,
+      nonce: 0,
+      initCode,
```

Comment:
Checking my understanding: won't it be a waste of gas to include this initcode with every l1 userop ?

Reply:
I think so, yes. You want to include it only if the wallet isn't already deployed.

Reply:
Good point! We'd pay slightly more gas due to the increased `callData` size of including the `initCode` if we included it with every UserOperation.